openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -332,7 +336,7 @@ def loadFile(self, files):
         self._XMLTrees = trees
 
         # Parse XML, get force definitions
-        # TODO: Use lazy instantiation to only parse XML trees when we need to.
+        # QUESTION: Should we use lazy instantiation to only parse XML trees when we need to so that users will manipulate parameters only via XML tags?
```

Comment:
Not sure I understand this. The lazy instantiation part sounds reasonable and would avoid the problem I'd worked around originally, where we change a tag and then have to re-parse the XML before creating the system.

However, I don't understand "so that users will manipulate parameters only via XML tags". Why would we want that? So far we provide an interface to allow people to change parameters, which has been valuable for us and for Bryce. Also, in discussions (e.g. at the January meeting) we talked about how we would make cutoffs, etc. be part of these files, but users doing testing would still be able to change these parameters via the API. If for some reason they have to edit the XML directly that would become a lot more cumbersome.

So, lazy instantiation, yes, but manipulate parameters only via XML tags, no.